low-level/files/opt/seagate/sspl/bin/generate_sspl_bundle/sspl_bundle_generate

```diff
@@ -30,77 +31,131 @@ from framework.base.sspl_constants import PRODUCT_FAMILY, SUPPORT_REQUESTOR_NAME
+        if not os.path.exists(self.localTempPath):
+            os.makedirs(self.localTempPath)
+        try:
+            sspl_state = ""
+            try:
+                sspl_state=open(f"/var/{PRODUCT_FAMILY}/sspl/data/state.txt").readline().rstrip().split("=")
+                # Capturing enclosure logs only when sspl state is active
+                sspl_state = sspl_state[1] if len(sspl_state)==2 else ''
+                if not sspl_state == 'active':
+                    print("SSPL is in 'degraded' mode, so enclosure data not collected as part of this node support bundle.")
 
-    configure_path = parser[2]
-    if configure_path:
-        if os.path.exists(configure_path):
-            localTempPath = configure_path+"sspl/"
-        else:
-            print("Given path doesn't exist")
+            except (FileNotFoundError, OSError) as e:
+                print(f"Failed to open the SSPL 'state.txt' file with an error '{e}',\
+                    Can't determine the SSPL state, So enclosure logs also being collected.")
+                sspl_state == 'active'
+
+            self.get_ipmi_sel_data()
+            self.get_consul_data()
+            self.get_drives_smart_data()
+            if sspl_state == "active" and "-noencl" not in parser:
+                self.get_enclosure_logs()
+                
+            with tarfile.open(self.localTempPath+file_name, "w:gz") as tar:
+                file_dir = file_name.replace(".tar.gz","")
+                if os.path.exists(sspl_log_dir):
+                    tar.add(sspl_log_dir, arcname=file_dir+"/sspl/logs/")
+                if os.path.exists(self.sspl_conf_dir):
+                    tar.add(self.sspl_conf_dir, arcname=file_dir+"/sspl/config/sspl.sls")
+                if os.path.exists(self.consul_sspl_config):
+                    tar.add(self.consul_sspl_config, arcname=file_dir+"/sspl/consul/consul_sspl_config.txt")
+                if os.path.exists(self.consul_sspl_data):
+                    tar.add(self.consul_sspl_data, arcname=file_dir+"/sspl/consul/consul_sspl_cache.txt")
+                if os.path.exists(self.boot_drvs_dta):
+                    tar.add(self.boot_drvs_dta, arcname=file_dir+"/drives_SMART_data/")
+                if os.path.exists(self.ipmi_sel_data):
+                    tar.add(self.ipmi_sel_data, arcname=file_dir+"/ipmi/ipmi_sel_data.txt")
+                if os.path.exists(self.rabbitmq_status):
+                    tar.add(self.rabbitmq_status, arcname=file_dir+"/rabbitmq/status/rabbitmq_status.txt")
+                if os.path.exists(self.rabbitmq_conf_dir):
+                    tar.add(self.rabbitmq_conf_dir, arcname=file_dir+"/rabbitmq/config/rabbitmq.config")
+                if os.path.exists(self.rabbitmq_log_dir):
+                    tar.add(self.rabbitmq_log_dir, arcname=file_dir+"/rabbitmq/logs/")
+                if os.path.exists(iem_log_dir):
+                    tar.add(iem_log_dir, arcname=file_dir+"/iems/")
+                tar.close()
```

Comment:
After tarring, make sure to cleanup temp files/folder created for generating support bundle

Reply:
Done

Reply:
line 318         shutil.rmtree(f"/tmp/{sspl_bundle_tmp}")
ok